src/main/java/com/google/sps/servlets/GoDataServlet.java

```diff
@@ -42,7 +45,15 @@
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     String indexStr = request.getParameter(INDEX_PARAMETER);
-    index = Integer.parseInt(indexStr);
+    try {
+      index = Integer.parseInt(indexStr);
+    } catch (Exception e) {
+    }
+
+    nameGuess = request.getParameter(NAME_PARAMETER);
```

Comment:
If I understand correctly, this design is that the client is expected to send only an index XOR a guess. Because those are two different pieces of functionality, they should be more clearly separated. I think separating them by having different URL paths for each would be the most simple and clear way of doing this*. I'm not sure how to configure a servlet to handle multiple different URL paths, but I imagine there's a way, otherwise you could write an additional servlet class for the different path. I can help you look for a way to do multiple paths in the same class later on if you'd like.

* Other ideas include having an additional parameter that specifies which set of functionality the client wants from the server for this request.

Reply:
This makes sense -- I think it'd be clearer if I split GoDataServlet into two separate servlets to handle getting the scavenger hunt data and getting the user's answer separately.